src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -885,11 +885,18 @@ public ConcurrentAsyncCallback(Action<T> handler, GraphStageLogic ownedStage)
                 _ownedStage = ownedStage;
                 _wrappedHandler = obj =>
                 {
-                    if (obj is T e)
-                        handler1(e);
-                    else
-                        throw new ArgumentException(
-                            $"Expected {nameof(obj)} to be of type {typeof(T)}, but was {obj.GetType()}");
```

Comment:
The NRE was thrown here by `obj.GetType()` when obj is null.

The actual fix was to make sure that this wrapper also handles null and passes it to the handler delegate